test/extended/openstack/loadbalancers.go

```diff
@@ -382,6 +383,152 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
 		})
 	}
+
+	// Only valid with the Octavia amphora provider. The OVN-Octavia provider does not
+	// support allowed_cidrs in the LB (https://issues.redhat.com/browse/OCPBUGS-2789)
+	g.It("should limit service access on an UDP Amphora LoadBalancer when an UDP LoadBalancer svc setting the loadBalancerSourceRanges spec is created on Openshift", func() {
+
+		// Skipped for OVN-Octavia provider
+		skipIfNotLbProvider("Amphora", cloudProviderConfig)
+
+		g.By("Creating Openshift deployment")
+		depName := "udp-lb-sourceranges-dep"
+		svcName := "udp-lb-sourceranges-svc"
+		svcPort := int32(8082)
+		replicas := int32(2)
+		labels := map[string]string{"app": depName}
+		drop_sourcerange := "9.9.9.9/32"
+		testDeployment := createTestDeployment(depName, labels, replicas, v1.ProtocolUDP, 8081)
+		deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
+			testDeployment, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By(fmt.Sprintf("Creating Openshift LoadBalancer Service with loadBalancerSourceRanges: '%s'", drop_sourcerange))
+		jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+		jig.Labels = labels
+		svc, err := jig.CreateLoadBalancerService(loadBalancerServiceTimeout, func(svc *v1.Service) {
+			svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: svcPort, TargetPort: intstr.FromInt(8081)}}
+			svc.Spec.Selector = labels
+			svc.Spec.LoadBalancerSourceRanges = []string{drop_sourcerange}
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Checks from openshift perspective")
+		loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+		o.Expect(loadBalancerId).ShouldNot(o.BeEmpty(), "load-balancer-id annotation missing")
+		e2e.Logf("LB ID: %s", loadBalancerId)
+		o.Expect(svc.Spec.LoadBalancerSourceRanges).Should(o.Equal([]string{drop_sourcerange}))
+		svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+		o.Expect(svcIp).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+		e2e.Logf("Service IP: %s", svcIp)
+
+		g.By("Checks from openstack perspective")
+		allPages, err := octavialisteners.List(loadBalancerClient, octavialisteners.ListOpts{LoadbalancerID: loadBalancerId}).AllPages()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		listeners, err := octavialisteners.ExtractListeners(allPages)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Check there is only one listener in the LB
+		o.Expect(listeners).Should(o.HaveLen(1), "More than one listener found for LB '%s': %v", loadBalancerId, listeners)
+		listenerId := listeners[0].ID
+		e2e.Logf("LB listener ID: %s", listenerId)
+
+		// Check allowed_cidrs in the LB listener matches with the svc spec
+		o.Expect(listeners[0].AllowedCIDRs).Should(o.ConsistOf([]string{drop_sourcerange}), "Unexpected allowed_cidrs in Openstack LoadBalancer Listener '%s'", listenerId)
```

Comment:
We should log the value of `listeners[0].AllowedCIDRs`. Will it log it?

Reply:
It's being logged when the gomega asserion fails. i.e (ignore the values again):
```
fail [/home/stack/openstack-test/test/extended/openstack/loadbalancers.go:458]: Unexpected allowed_cidrs in Openstack LoadBalancer Listener '8aa7c328-540b-4215-b79d-1ddbb9716cb7'
Expected
    <[]string | len:1, cap:4>: ["9.9.9.9/32"]
to consist of
    <[]string | len:1, cap:1>: ["10.10.10.10/32"]
the missing elements were
    <[]string | len:1, cap:1>: ["10.10.10.10/32"]
the extra elements were
    <[]string | len:1, cap:1>: ["9.9.9.9/32"]
Ginkgo exit error 1: exit with code 1
```